internal/search/codeownership/job.go

```
@@ -73,15 +86,49 @@ func (s *codeownershipJob) MapChildren(fn job.MapFunc) job.Job {
 	return &cp
 }
 
-func applyCodeOwnershipFiltering(ctx context.Context, includeOwners, excludeOwners []string, matches []result.Match) ([]result.Match, error) {
+func applyCodeOwnershipFiltering(
+	ctx context.Context,
+	db database.DB,
+	mu *sync.Mutex,
+	rules *map[string]Ruleset,
+	includeOwners,
+	excludeOwners []string,
+	matches []result.Match) ([]result.Match, error) {
+	var errs error
+
 	filtered := matches[:0]
 
-	// We currently don't have a way to access file ownership information, so no
-	// file currently has any owner. A search to include any owner will
-	// therefore return no results.
-	if len(includeOwners) == 0 {
-		filtered = matches
+matchesLoop:
+	for _, m := range matches {
+		switch mm := m.(type) {
+		case *result.FileMatch:
+			cachekey := fmt.Sprintf("%s@%s", mm.Repo.Name, mm.CommitID)
+
+			mu.Lock()
+			ruleset, ok := (*rules)[cachekey]
+			if !ok {
+				var err error
+				ruleset, err = NewRuleset(db, mm.Repo.Name, mm.CommitID)
+				if err != nil {
+					errs = errors.Append(errs, err)
+				}
+				(*rules)[cachekey] = ruleset
+			}
+			owners, _ := ruleset.Match(mm.File.Path)
```

Why ignore the error here?